Handcuff balance changes, Ruko edition!

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This pull request tweaks the order of operations with attempts to break out of cuffs, and also adds a dash of RNG into the mix as well. For most breakout attempts this RNG will not factor. 

Specifically what is changed:
* Players who are **bucklecuffed** will work their way out of the cuffs first now, instead of taking a full minute or more to unbuckle themselves only to still be cuffed. Buckling someone who is cuffed will prevent them from running around, but will not make it take any additional time for them to break out of their restraints, and once the restraints are off they can easily unbuckle themselves.

* Players who are restrained in any way (handcuffs, bolas/legcuffs, straight jacket) are no longer forced to stand still in order to begin working their way out of the restraints. This means the "keep dragging them in circles" is no longer going to do anything to slow a potential escape attempt.

* Because of the above bullet point dramatically reducing the practical time it takes for a player to break out of restraints, all handcuff type restraints have had their base breakout time increased substantially to compensate. Handcuffs take three minutes, zip ties take two minutes and cable cuffs take one minute.

* Rather than needing to attempt a breakout for the full duration all at once, players will loop through a 5-second interval. Every 5 second interval brings the player closer to breakout, so even if they are stunned again in the middle of their escape attempt they don't lose all of their progress. Once the player has spent a total number of 5-second attempts equal to the breakout time, they will be freed from the cuffs. 

* And finally the sprinkles of RNG: every breakout attempt adds a 0.25% chance of successfully wriggling out sooner, and rather than squirming out of cuffs always being blatantly obvious, each 5-second attempt offers only a small chance of being noticed by nearby players. 

And then a few assorted tweaks because I was in the area of code anyway:
* Updated several numbers still in deciseconds to use the MINUTES and SECONDS macros
* Adjusted the straight jacket down from five minutes to only two minutes for breaking out
* Removed the "instant cuff break" for xenomorphs, they now use the same one as hulks do since they can break the cuffs while moving
* Breakout click cooldown has been reduced from 10 seconds to 2 seconds, because a 10 second click cooldown is such a long time it leads to players believing they are experiencing a bug when it's actually working entirely as intended. There's no reason to put players on a 10 second click cooldown, though this will make breaking out of certain objects (such as welded lockers) take a lot less time. I don't think this time difference is consequential in the vast majority of cases though. 


## Why It's Good For The Game

First and foremost because there is nothing worse as a player than being unable to do or interact with anything for extended periods of time. If security is parading someone around for an extended period of time instead of processing them, it really sucks for that player and they do not often remember to include time spent being paraded around as time served. In many cases it can be worse than being dead because you can't even chill with ghosts or decide if there's another role you would like to take. 

Handcuffs are a good tool for restraining players, but they should not be an indefinite restraint for an officer to endlessly parade someone around, and players deserve more opportunities to keep going after non-lethal takedowns. Players exercising their chances at freedom should be aware that if they are not possible to stop with non-lethal force, security may be forced to resort to more lethal implements instead. 

If security notices a breakout attempt, they can stun, remove and re-apply the cuffs to "re-adjust" them and reset the breakout counters, so this does not completely remove the ability for security to keep someone restrained, but does add more risk to doing so for extended periods of time instead of processing the detained player. 

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

This is me managing to break out at 40 seconds, getting very lucky on the compounding chance
![dreamseeker_eYFZzYldZp](https://github.com/user-attachments/assets/d431d960-e257-4c80-8bc9-3b813861fe0d)

Testing the bucklecuffing:
![image](https://github.com/user-attachments/assets/320f96d4-64ad-4e5d-8cf1-d29565811444)

I tested specifically that the resist buttons and interface buttons both work to remove the cuffs, both when buckled and when not, as well as testing that I can move around while trying to remove cuffs. 

## Changelog
:cl:
balance: Hand restraints such as handcuffs are now much simpler to break out of. Each attempt takes only five seconds and gives a miniscule chance of breaking out immediately if you get lucky with a compounding chance after each attempt. Each attempt stacks up to eventually being enough to guarantee the cuffs are removed once your total attempts are equal to the breakout time as well. 
balance: Attempts to remove any types of restraint (Bola, cuffs, straight jackets, etc.) no longer require you to be still and motionless, which means you can run around like a headless chicken and still break out of some cuffs. Being dragged in circles by an officer no longer interrupts breakout attempts either.
balance: Because the general process for breaking hand restraints is now much more consistent, the base times for all three hand restraints has been increased dramatically so that handcuffs are still effective at their intended purpose, just without being an indefinite restraint.
tweak: Buckle-cuffed players are able to remove their restraints first now, so buckling a player who is cuffed does nothing to lengthen the effective restraint time further, it only prevents them from running around. 
tweak: straight jackets take two minutes to break out of instead of five now, because five is an excessive amount of time to potentially stack on top of cuffs. 
tweak: Xenomorphs can no longer instantly break cuffs. They use the same define and code as hulks do, taking five seconds to break the cuffs. In both hulk and xeno cases, the cuffs can now be broken while moving. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
SECONDS and MINUTES

Rukofamicom committed:
Restraints before buckle

Rukofamicom committed:
move with cuff breaking, no chat message

Rukofamicom committed:
fix chat message

Rukofamicom committed:
Cuff attempts are 5 seconds and stack

Rukofamicom committed:
whoops, didn't push part of that last one

Tsar-Salat commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -295,25 +295,35 @@ CREATION_TEST_IGNORE_SELF(/mob/living/carbon)
+		to_chat(src, span_notice("You attempt to wriggle your way out of [cuffs]..."))
+		while(do_after(src, 5 SECONDS, timed_action_flags = IGNORE_USER_LOC_CHANGE|IGNORE_HELD_ITEM, hidden = TRUE))
+			cuff_breakout_attempts++
+			if(cuff_breakout_attempts * 5 SECONDS >= breakouttime || prob(cuff_breakout_attempts))
```
message_admins("[src] broke out of [cuffs] after [cuff_breakout_attempts]")

Or somethin.

Rukofamicom replied:
I don't mind but I do worry about some spam

Rukofamicom replied:
included in most recent push where I tweaked the math on breakout

Rukofamicom committed:
Whoops, forgot the last push. Glad nobody has code reviewed yet.

Rukofamicom committed:
Update carbon.dm

Rukofamicom committed:
Update carbon.dm

Tsar-Salat reviewed (dismissed):
This is another log request, but can you add a log for when handcuffs are taken off? Theres a combat log for when they are applied, but thats it.

Im handling a report right now and I realize that its not logged which is quite annoying.

I will give my approval afterwards.

Rukofamicom committed:
moves to combat log

Rukofamicom committed:
Removes unnecessary check causing inconsistent delays

Rukofamicom committed:
Reduces CLICK_CD_BREAKOUT to two seconds

No reason to have a ten second cooldown on breakout attempts. This will make breaking out of welded lockers and such easier though.

Rukofamicom committed:
Order of operations error + resist button bug

Rukofamicom committed:
Revert unnecessary changes to living.dm

Rukofamicom committed:
Merge branch 'BeeStation:master' into Cuffs

Rukofamicom committed:
Fixes bucklecuff breakout

Rukofamicom committed:
Merge branch 'Cuffs' of https://github.com/Rukofamicom/BeeStation-Hornet into Cuffs
